impeller/entity/geometry.cc

```diff
@@ -16,6 +16,46 @@
 
 namespace impeller {
 
+/// Given a convex polyline, create a triangle fan structure.
+std::pair<std::vector<Point>, std::vector<uint16_t>> TessellateConvex(
+    Path::Polyline polyline) {
+  std::vector<Point> output;
+  output.reserve(polyline.points.size() * 3);
+
+  for (auto j = 0u; j < polyline.contours.size(); j++) {
+    Scalar center_x = 0;
+    Scalar center_y = 0;
+    auto [start, end] = polyline.GetContourPointBounds(j);
+    for (auto i = start; i < end; i++) {
+      const auto& point = polyline.points[i];
+      center_x += point.x;
+      center_y += point.y;
+    }
+    center_x /= end - start;
+    center_y /= end - start;
+
+    for (auto i = start + 1; i < end; i++) {
+      const auto& point_a = polyline.points[i - 1];
+      const auto& point_b = polyline.points[i];
+      output.emplace_back(point_a);
+      output.emplace_back(point_b);
+      output.emplace_back(Point(center_x, center_y));
+    }
+    // Some shapes don't seem to self close?
+    if (polyline.points[end - 1] != polyline.points[start]) {
+      output.emplace_back(polyline.points[end - 1]);
+      output.emplace_back(polyline.points[start]);
+      output.emplace_back(Point(center_x, center_y));
+    }
+  }
+
+  std::vector<uint16_t> index(output.size());
+  for (auto i = 0u; i < output.size(); i++) {
+    index[i] = i;
+  }
```

Comment:
We should drop this and just use `kTriangleFan`. We need to teach the rest of the pipeline to not require an index buffer in that case.

Reply:
(Talked offline, I guess Metal removed this, so ok - instead just copy the vertices to the vertex buffer and make the index buffer form them into a strip).